app_backend/database.py

```diff
@@ -41,10 +41,16 @@ def save_user_in_db(self, login_info: dict):
         if not self.test_mode:
             print("[Database.py] Dry run not updating database")
             login_info["verified"] = False
+            login_info["friends"] = []
             users.insert_one(login_info)
 
         return {"msg": "Successfully registered new user."}
 
+    def get_user_by_uuid(self, uuid):
+        users = self.db["users"]
+        return users.find_one({"_id": uuid}, {"password": 0})
```

Comment:
For this part, you are searching up the user with the uuid, why include the password? Are you checking that they don't have a password?

Reply:
This function returns user data -- but since we are only getting that data from the UUID, we don't want to include sensitive information. We want to use this function for something like getting the friends of a user. We don't want to include any sensitive information from the friend. So`{"password" : 0}` excludes the `password` from the result

Reply:
ohhh gotcha, very smart